### ISSUE TITLE
Simple Data-Binding-Only flavor of Aurelia 2

### ISSUE THREAD

**Issue by nenadvicentic** [aurelia/aurelia#1312]
Frequent scenario in web site development: enhancement (hydration) of a part of an HTML document pre-rendered by a server, is currently partially supported by Aurelia 2.

For this type of scenario it would be important to have:

1. Quick and simple setup, ideally just referencing correct package on CDN.
2. Possibility to work with browser's JavaScript alone, without need for a setup of CLI, transpilers, bundlers, NodeJS, etc...
3. Good startup performance. Bundled and minified package would significantly decrease network latency on first load.

Simplicity would help introduce new developers to Aurelia and decrease complexity in projects where full blown framework is not needed.

Currently, it's possible to reference native ES module version of `@aurelia/runtime-html` package via CDN and wire-up data binding.

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Simple Data-Binding</title>
</head>
<body>
    <div id="my-app">
        <label>
            Message:
            <input type="text" class="form-control" value.bind="message" />
        </label>
        <button class="btn btn-primary" click.trigger="reverseMessage()">Reverse Message</button>
        <p>
            ${ message }
        </p>
    </div>
    <script type="module">
        import { Aurelia, StandardConfiguration } from 'https://cdn.jsdelivr.net/npm/@aurelia/runtime-html@2.0.0-alpha.21/dist/native-modules/index.js';

        class MyApp {
            message = "Hello Aurelia 2!";

            messageChanged(newValue, oldVelue) {
                console.log(`${oldVelue} -> ${newValue}`);
            }

            reverseMessage() {
                this.message = this.message.split('').reverse().join('');
            }
        };

        new Aurelia()
            .register(StandardConfiguration)
            .enhance({
                host: document.querySelector('#my-app'),
                component: MyApp
            });
    </script>
</body>
</html>
```

Gist example is [here](https://gist.github.com/nenadvicentic/7bcc8a7a608f9481d009344d03634ef0).

However, there are a few missing features with this example.

### First page-load performance
When page gets loaded for a first time and script have not been cached, there is significant lag between initial page render and moment when Aurelia data-binding grabs rendered HTML element and replaces `${message}` placeholder.

Reason is that browser has to fetch multiple related script modules via network (platform, platform-browser, kernel, runtime, metadata) and it adds up to the delay. Bundled + minified version would have significantly faster startup time.

### Wiring up `messageChanged`  handler
Normally, to configure `messageChanged` handler to be executed on `message` property changes, we would add `@observable` decorator on the property.

```typescript
@observable
message = "Hello Aurelia 2!";
```

Decorators are currently only a TypeScript experimental feature, so browser does not know how to wire them up. In Aurelia v1, there was a static `Decorate` helper method (or similar name, can't remember exactly) to help working with decorators from ES5 directly. I'm not aware if anything similar exists in Aurelia 2.

**Comment by timfish** [comment]
`aurelia` imports `@aurelia/fetch-client`,`@aurelia/kernel`,`@aurelia/metadata`,`@aurelia/platform`,`@aurelia/platform-browser`,`@aurelia/route-recognizer`,`@aurelia/router`,`@aurelia/runtime` and `@aurelia/runtime-html`.

You can already exclude the router, fetch, etc by using specific packages rather than the top level `aurelia`.

This could be made easier with modules by adding a package `export` in `package.json` (maybe under `aurelia/minimal`?) that doesn't include `@aurelia/fetch-client`,`@aurelia/route-recognizer` and `@aurelia/router`. 

Are there any CDNs that will bundle ESM?

**Comment by nenadvicentic** [comment]
> You can already exclude the router, fetch, etc by using specific packages rather than the top level `aurelia`.

In my initial example I already used `@aurelia/runtime-html` and not full `aurelia`, as initially suggested by @bigopon on [discourse thread](https://discourse.aurelia.io/t/app-with-just-aurelia-binding-and-templating/841/15) . This eliminates extra modules/packages.

What to me seems a bigger issue is, how to use TypeScript decorators within native JavaScript? Perhaps `aurelia/minimal` would have to expose few addition helper methods? To me, open question is how to use TypeScript decorators within native JavaScript.  Use of `@observable` is such common case that, without it working, `aurelia/minimal` would not be very useful.

**Comment by timfish** [comment]
You can't use TypeScript decorators in JavaScript. TypeScript uses the legacy decorator spec that never made it into JavaScript. As far as I understand it, the new decorator spec hasn't made it to stage 3 and isn't supported anywhere yet.

Aurelia allows decorators OR static properties as an alternative. I'm on mobile at the moment and can't find an example but it's all in the docs.

**Comment by nenadvicentic** [comment]
> You can't use TypeScript decorators in JavaScript.

Yes, it's clear. I already wrote that in detail above. TypeScript emits `__decorate` and `__metadata` helper functions to support decorators functionality.

> Aurelia allows decorators OR static properties as an alternative.

I'm aware of such feature, static properties, in Aurelia v1 (can't find it at the moment in docs). In Aurelia v2 I did not find anything similar so far.

**Comment by timfish** [comment]
![Screenshot_20211107-092820.jpg](https://user-images.githubusercontent.com/1150298/140639639-ed43c01e-34cb-4397-82ea-62108957c038.jpg)

The DI docs actually cover static property before decorators

**Comment by nenadvicentic** [comment]
@timfish I think you have to read the issue and subsequent reply with example code more carefully.

Documentation you provided solves only one specific situation, replacement for `@inject` decorator. This does not help with property decorator `@observe`, or with decorators in general.

Main question right now is - how to wire up `${propertyName}Chanded` from plain JavaScript?

**Comment by fkleuver** [comment]
The main use case is now covered by Aurelia’s no-build browser setup, ESM CDN imports, `CustomElement.define`, and `.enhance()`, with a vanilla JavaScript example available as well.

We never created a dedicated global “binding-only” bundle, but the original browser and tooling constraints have changed substantially. I’m closing this as addressed. If the current no-build route still has a concrete size or startup-performance problem, please open a new issue with recent measurements.
